syft/pkg/cataloger/ai/cataloger.go

```diff
@@ -5,12 +5,165 @@ including support for GGUF (GPT-Generated Unified Format) model files.
 package ai
 
 import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+
+	gguf_parser "github.com/gpustack/gguf-parser-go"
+
+	"github.com/anchore/syft/internal"
+	"github.com/anchore/syft/internal/log"
+	"github.com/anchore/syft/syft/artifact"
+	"github.com/anchore/syft/syft/file"
 	"github.com/anchore/syft/syft/pkg"
 	"github.com/anchore/syft/syft/pkg/cataloger/generic"
+	"github.com/anchore/syft/syft/source/ocimodelsource"
+)
+
+const (
+	catalogerName = "gguf-cataloger"
 )
 
+// ggufCataloger implements pkg.Cataloger with support for both file-based and OCI layer-based discovery.
+type ggufCataloger struct {
+	genericCataloger pkg.Cataloger
+}
+
 // NewGGUFCataloger returns a new cataloger instance for GGUF model files.
+// It supports both traditional file-based discovery and OCI layer-aware discovery
+// when the resolver implements OCIResolver.
 func NewGGUFCataloger() pkg.Cataloger {
-	return generic.NewCataloger("gguf-cataloger").
-		WithParserByGlobs(parseGGUFModel, "**/*.gguf")
+	return &ggufCataloger{
+		genericCataloger: generic.NewCataloger(catalogerName).
+			WithParserByGlobs(parseGGUFModel, "**/*.gguf"),
+	}
+}
+
+// Name returns the cataloger name.
+func (c *ggufCataloger) Name() string {
+	return catalogerName
+}
+
+// Catalog discovers GGUF model packages from the given resolver.
+// If the resolver implements OCIResolver, it uses layer-aware discovery.
+// Otherwise, it falls back to glob-based file discovery.
+func (c *ggufCataloger) Catalog(ctx context.Context, resolver file.Resolver) ([]pkg.Package, []artifact.Relationship, error) {
+	// Check if the resolver supports OCI layer-aware access
+	if ociResolver, ok := resolver.(ocimodelsource.OCIResolver); ok {
+		log.Debug("using OCI layer-aware discovery for GGUF models")
+		return c.catalogFromOCILayers(ctx, ociResolver)
+	}
+
+	// Fall back to generic glob-based discovery
+	log.Debug("using glob-based discovery for GGUF models")
+	return c.genericCataloger.Catalog(ctx, resolver)
+}
+
+// catalogFromOCILayers discovers GGUF models by querying OCI layers by media type.
+func (c *ggufCataloger) catalogFromOCILayers(ctx context.Context, resolver ocimodelsource.OCIResolver) ([]pkg.Package, []artifact.Relationship, error) {
+	// Find all GGUF layers by media type
+	digests, err := resolver.LayerDigestsByMediaType(ocimodelsource.GGUFLayerMediaType)
```

Comment:
we shouldn't import from a package meant for a specific implementation of a source